src/mpi/comm/comm_split_type.c

```diff
@@ -474,10 +479,45 @@ static int get_color_from_subset_bitmap(int node_index, int *bitmap, int bitmap_
 
 }
 
-static int network_split_by_minsize(MPIR_Comm * comm_ptr, int key, int subcomm_min_size,
-                                    MPIR_Comm ** newcomm_ptr)
+static void get_prime_factors(int num, int node_dimensions, int **prime_factors, int *num_factors)
```

Comment:
We have a function to get prime factors in `MPI_Dims_create` too.  Maybe we can merge them into a single function in `MPL_`?

Reply:
Will do.